include/envoy/event/dispatcher.h

```diff
@@ -39,6 +57,14 @@ class Dispatcher {
    */
   virtual TimeSource& timeSource() PURE;
 
+  /**
+   * Start writing stats once thread-local storage is ready to receive them (see
+   * ThreadLocalStoreImpl::initializeThreading).
+   * @param parent_scope the parent scope for the new per-dispatcher stats scope created here.
+   * @param name the name for the new stats scope, identifying this dispatcher.
```

Comment:
What happens for stats for dispatchers that are allocated and run without calling this? 

I think this might be OK but just specify it here. Auto-name them?

Reply:
What I've put in the docs, at least for now, is that we only record event loop stats for the main thread and workers, not any other auxiliary threads. I assume other threads might be transient, and it would just be noisy to include stats for them automatically. What do you think?

Reply:
I think that's OK, but can we (a) document that here and (b) have a stat for the number of anonymous untracked threads with dispatchers?

You can also survey source/.../*.cc for references to allocateDispatcher(). There's at least one you should probably add, in server/guarddog_impl.cc